drivers/periph_common/Kconfig.timer

```diff
@@ -10,7 +10,7 @@ menuconfig MOD_PERIPH_TIMER
     depends on HAS_PERIPH_TIMER
     select MOD_PERIPH_COMMON
 
-if MOD_PERIPH_TIMER
+if MOD_PERIPH_TIMER && TEST_KCONFIG
```

Comment:
The dependency to `TEST_KCONFIG` should be applied to `MOD_PERIPH_TIMER` as well.